framework/tlvf/tlvf.py

```diff
@@ -726,6 +731,8 @@ def addClassParam(self, obj_meta, param_name, param_type, param_type_info, param
             if is_var_len:
                 # variable length list support - swap length value for calculating list size
                 length_type = obj_meta.children_types[param_length]
+                param_meta.list_index = obj_meta.list_index
```

Comment:
This mechanism should also be added to dynamic lists (is_dynamic_len - unknown list length).